### PR TITLE
Fix file mode handling for zip entries with non-Unix platform

### PR DESCRIPTION
**Pull request by plamentotev**
Let's say you have a zip archive created under windows or with tool that does not support unix files modes. `org.apache.commons.compress.archivers.zip.ZipArchiveEntry::getUnixMode` will return `0` for any entry that does not have platform set to Unix.

On the other hand `ZipResource::getAttributes` does not check for this case and will return `0` (this is valid Unix mode).

As a result if you add zip archive using `AbstractArchiver::addArchivedFileSet` the resulting archive will have entries with file mode set to `0`. That's quite bad because when unpacked on some Unix platforms the files will be accessible only by the root user.

Instead `ZipResource::getAttributes` should return `-1` so the rest of plexus-archiver code could handle the case properly and distinguish it from the `0` file mode.

Add check in the `ZipResource::getAttributes` to properly handle non-Unix platform entries.

### CODE REVIEW

**Comment by plamentotev** [comment]
This time I didn't forgot `PlexusIoZipFileResourceCollection` :smile: it's just that `PlexusIoZipFileResourceCollection` does not implement `ResourceAttributeSupplier` and it does not gets the zip entries' mode.

File: src/main/java/org/codehaus/plexus/archiver/zip/ZipResource.java [hunk 1]
```diff
@@ -45,14 +45,15 @@ private static long getLastModofied( ZipArchiveEntry entry )
 
     public synchronized PlexusIoResourceAttributes getAttributes()
     {
-        int mode = entry.getUnixMode();
-        if ( ( mode & UnixStat.FILE_FLAG ) == UnixStat.FILE_FLAG )
+        int mode = -1;
```
Comment (plamentotev) [inline src/main/java/org/codehaus/plexus/archiver/zip/ZipResource.java]: I think it's better to use constant. But didn't found existing one? Maybe we can introduce new one in `PlexusIoResourceAttributes`. For example `UNKNOWN_OCTAL_MODE`

Comment (krosenvold) [inline src/main/java/org/codehaus/plexus/archiver/zip/ZipResource.java]: Yes please, make a constabt
21. feb. 2016 16.08 skrev "Plamen Totev" <notifications@github.com>:

Comment (plamentotev) [inline src/main/java/org/codehaus/plexus/archiver/zip/ZipResource.java]: Done. I've just created a pull request - codehaus-plexus/plexus-io#3

**Comment by plamentotev** [comment]
Thanks for accepting the pull request, but I'm a bit confused now. I've introduced the `PlexusIoResourceAttributes.UNKNOWN_OCTAL_MODE` in [plexus-io](https://github.com/codehaus-plexus/plexus-io) but when to use it here? I guess I have to wait for a pexus-io release and when the dependency version is updated to 2.7.1 to use it in plexus-archiver. I mean there is no need to make a release for such a small change, right?

**Comment by krosenvold** [comment]
I will be releasing io soonish :)
23. feb. 2016 22.16 skrev "Plamen Totev" <notifications@github.com>:

**Pull request merged**